Aptos chain support from sdk

karansinglacoindcx opened the pull request.

akathecoder commented on packages/core-js/src/userop/index.ts:
```diff
@@ -1,3 +1,6 @@
-export { evmRawTransaction } from './rawTransaction.js';
+export { evmRawTransaction } from './evmRawTransaction.js';
 export { tokenTransfer } from './tokenTransfer.js';
+export { nftCreateCollection } from './nftCollectionCreation.js';
```
@SrijanCoinDCX they are working from Backend ?

akathecoder commented on packages/core-js/src/utils/constants.ts:
```diff
@@ -25,8 +25,9 @@ export class Constants {
     TOKEN_TRANSFER: 'TOKEN_TRANSFER',
     NFT_TRANSFER: 'NFT_TRANSFER',
-    NFT_COLLECTION_CREATION: 'NFT_COLLECTION_CREATION',
+    NFT_COLLECTION_CREATION: 'NFT_CREATE_COLLECTION',
```
name it same as intent on chain

akathecoder replied:
it is still not resolved @karansinglacoindcx

akathecoder replied:
+ @SrijanCoinDCX

akathecoder commented on packages/core-js/src/userop/aptosRawTransaction.ts:
```diff
@@ -0,0 +1,149 @@
+    });
+  }
+
+  const transactionsBytes = data.transactions.map((transaction) => {
+    const aptosTransaction: AptosRawTransaction = {
+      function: transaction.function,
+      typeArguments: transaction.typeArguments || [],
+      functionArguments: transaction.functionArguments || [],
+    };
+    return toHex(stringToBytes(JSON.stringify(aptosTransaction)));
+  });
+
+  const jobparam = encodeAbiParameters(
+    parseAbiParameters(jobParametersAbiType),
+    [
+      {
+        caip2Id: data.caip2Id,
+        transactions: transactionsBytes,
+      },
+    ],
+  );
```
@SrijanCoinDCX can Aptos Raw Txn execute multiple txns in same intent?
If no, please remove array as an input from function, make it array in calldata generation only

akathecoder replied:
@karansinglacoindcx whats the update here ?

karansinglacoindcx replied:
have made it array in calldata generation itself

akathecoder commented on packages/core-js/src/userop/types.ts:
```diff
@@ -36,23 +36,81 @@ export type NFTTransferIntentParams = {
+
+export interface AptosRawTransactionIntentParams {
+  caip2Id: string;
+  transactions: AptosRawTransaction[];
```
@SrijanCoinDCX same, check with Backend Team and update accordingly.

akathecoder commented on packages/core-js/src/userop/nftMint.ts:
```diff
@@ -0,0 +1,149 @@
+    throw new BaseError('User not logged in');
+  }
+
+  validateSchema(NftMintParamsSchema, data);
```
NFTMint and NftCollectionCreation is Aptos specific intents for now, please confirm with BE team and add a check for same

akathecoder replied:
aptosRawTransaction as well

karansinglacoindcx commented:
This pull request introduces significant enhancements to the SDKs by adding support for Aptos raw transactions, NFT minting, and NFT collection creation. It also includes refactoring and file renaming to improve clarity and maintainability. Below are the most important changes grouped by theme.

### New Features
* Added a new method, `aptosRawTransaction`, to support raw transactions on the Aptos blockchain. This includes creating user operations, signing them, and executing them via `OktoClient`. (`packages/core-js/src/abstract/aptosRawTransaction.ts`, `packages/core-js/src/userop/aptosRawTransaction.ts`) [[1]](diffhunk://#diff-e363af3d33938d9c85bd0043839e0c2333ebd567af6d446924e17802703c424eR1-R24) [[2]](diffhunk://#diff-971594621dd3a84453fb9037764f74e7efd4101eef224b37628047eb4faf1b54R1-R149)
* Introduced `nftMint` to enable minting NFTs on the Aptos blockchain. This includes encoding NFT minting parameters and creating user operations. (`packages/core-js/src/abstract/nftMint.ts`, `packages/core-js/src/userop/nftMint.ts`) [[1]](diffhunk://#diff-6b05b77c8a3bdf57d6d72fc960b7ce968935b339e2c41a0446f5568ea8649b40R1-R24) [[2]](diffhunk://#diff-6109f8b12b630d9bb004e6fc652d52b01ea1bea9b9c061dba8c8e78edc4223b0R1-R155)
* Added `nftCreateCollection` for creating NFT collections on the Aptos blockchain. This includes validation, encoding collection parameters, and generating user operations. (`packages/core-js/src/abstract/nftCollectionCreation.ts`, `packages/core-js/src/userop/nftCollectionCreation.ts`) [[1]](diffhunk://#diff-2df2093f1e11cb9b18d14d83b158ad187f2dbec9a97aa11eacc12e88bd596e21R1-R24) [[2]](diffhunk://#diff-eb522a13f15bec26f3dc39700d5f1298403d24429f6758f4eea0923c4f31fdaeR1-R3)

### Refactoring and File Organization
* Renamed `rawTransaction.ts` to `evmRawTransaction.ts` and updated the method `evmRawTransaction` to include stricter validation for EVM-compatible chains. (`packages/core-js/src/abstract/evmRawTransaction.ts`, `packages/core-js/src/userop/evmRawTransaction.ts`) [[1]](diffhunk://#diff-252f6896c7ecd4c553484d43239bf51e5882371f9c2cd659326c86e73e38e291L2-R10) [[2]](diffhunk://#diff-75fbbc8a6726bff3212f4377809d422da205587dc35115215f6d4fefd1a7fc4dL17-R22) [[3]](diffhunk://#diff-75fbbc8a6726bff3212f4377809d422da205587dc35115215f6d4fefd1a7fc4dR67-R72)
* Updated exports in `index.ts` files to reflect the new methods and file renaming. (`packages/core-js/src/abstract/index.ts`, `packages/core-js/src/userop/index.ts`) [[1]](diffhunk://#diff-e068272dbbe39309f496d0ceffe1429744096d97f6a971ac4cc755b1e5d61d37L2-R6) [[2]](diffhunk://#diff-c57a2b26ee48bc54976dc035335e8c97aea9b9ca80f6248c93240e85b38c5d44L2-R6)

### Metadata and Release Notes
* Added a changeset file to document the new features and specify minor version bumps for `@okto_web3/react-native-sdk`, `@okto_web3/core-js-sdk`, and `@okto_web3/react-sdk`. (`.changeset/cold-kiwis-drive.md`)
